tests/core/pm-module/conftest.py

```diff
@@ -0,0 +1,260 @@
+    strategy = solidity_registry_strategy()
+    registry_package = Package(manifest, w3)
+    registry_deployer = Deployer(registry_package)
+    registry_deployer.register_strategy("PackageRegistry", strategy)
+    deployed_registry_package = registry_deployer.deploy("PackageRegistry")
+    assert isinstance(registry_package, Package)
+    registry = deployed_registry_package.deployments.get_instance("PackageRegistry")
+    return SolidityReferenceRegistry(to_canonical_address(registry.address), w3)
+
+
+def vy_registry(w3):
+    registry_path = ASSETS_DIR / "vyper_registry"
+    manifest = twig_manifest(registry_path, "registry", "1.0.0")
+    registry_package = Package(manifest, w3)
+    registry_deployer = Deployer(registry_package)
+    deployed_registry_package = registry_deployer.deploy("registry")
+    registry_instance = deployed_registry_package.deployments.get_instance("registry")
+    assert registry_instance.functions.owner().call() == w3.eth.defaultAccount
+    return VyperReferenceRegistry(to_canonical_address(registry_instance.address), w3)
+
+
+def release_packages(registry):
+    registry._release(
+        "package", "1.0.0", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGV"
+    )
+    registry._release(
+        "package", "1.0.1", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGW"
+    )
+    registry._release(
+        "package", "1.0.2", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGX"
+    )
+    registry._release(
+        "package", "1.0.3", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGJ"
+    )
+    registry._release(
+        "package", "1.0.4", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGK"
+    )
+    registry._release(
+        "package", "1.0.5", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGH"
+    )
+    registry._release(
+        "package1", "1.0.1", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGZ"
+    )
+    registry._release(
+        "package2", "1.0.1", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGT"
+    )
+    registry._release(
+        "package3", "1.0.0", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGA"
+    )
+    registry._release(
+        "package4", "1.0.0", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGB"
+    )
+    registry._release(
+        "package5", "1.0.0", "ipfs://Qme4otpS88NV8yQi8TfTP89EsQC5bko3F5N1yhRoi6cwGC"
+    )
+    return registry
+
+
+W3 = w3()
+FRESH_VY_REGISTRY = vy_registry(W3)
+FRESH_SOL_REGISTRY = sol_registry(W3)
+LOADED_VY_REGISTRY = release_packages(vy_registry(W3))
+LOADED_SOL_REGISTRY = release_packages(sol_registry(W3))
+VY_PKG_IDS = (VY_PACKAGE_ID_1, VY_PACKAGE_ID_2, VY_PACKAGE_ID_3)
+SOL_PKG_IDS = (SOL_PACKAGE_ID_1, SOL_PACKAGE_ID_2, SOL_PACKAGE_ID_3)
+VY_RLS_IDS = (VY_RELEASE_ID_1, VY_RELEASE_ID_2, VY_RELEASE_ID_3, VY_RELEASE_ID_4)
+SOL_RLS_IDS = (SOL_RELEASE_ID_1, SOL_RELEASE_ID_2, SOL_RELEASE_ID_3, SOL_RELEASE_ID_4)
+
+
+@pytest.fixture
+def w3():
+    return W3
```

Comment:
I wend with this structure simply to speed up the tests. 
- module level variables ~ 100seconds
- fixtures as you suggested ~ 1000seconds
I also tried scoping via `@pytest.fixture(scope="session")` - but that requires that I similarly scope `vy_deployer` and `sol_deployer` in `pytest-ethereum`. I didn't think this was a great idea, since those plugins are going to be mostly hidden when used in other libraries, and scoping them didn't seem like a great idea.
I designed the tests in a way that they all expect leakage from the other tests sharing the module-level variables, so maybe it's worth including a note on this setup? Or if you think it's worth the inter-test separation as a trade-off for the speed of running the tests, I'm also open to that.

Reply:
I think it's worth a comment at minimum denoting that the design is intentional.